isolate.sh

```diff
@@ -526,6 +539,89 @@ function cleanup_webhook() {
+        if [ $check_retries -eq 0 ]; then
+            error "Cert manager webhook pod not found in namespace $namespace"
+        fi
+    done
+    condition="oc -n ${namespace} get po --no-headers --ignore-not-found | egrep 'Running|Completed|Succeeded' | grep ^${name}"
+    wait_message="Waiting for pod ${name} in namespace ${namespace} to be running ..."
+    success_message="Pod ${name} in namespace ${namespace} is running."
+    error_message="Timeout after ${total_time_mins} minutes waiting for pod ${name} in namespace ${namespace} to be running."
+    wait_for_condition "${condition}" ${retries} ${sleep_time} "${wait_message}" "${success_message}" "${error_message}"
+    
+    #controller
+    name=$(${OC} get pod -n $namespace | (grep cert-manager-controller || echo "fail") | awk '{print $1}')
+    check_retries=10
+    while [[ $name == "fail" ]] && [ $check_retries -gt 0 ]; do
+        sleep 15
+        name=$(${OC} get pod -n $namespace | (grep cert-manager-controller || echo "fail") | awk '{print $1}')
+        debug1 "cert manager controller pod present: $name"
+        check_retries=$(( check_retries - 1 ))
+        if [ $check_retries -eq 0 ]; then
+            error "Cert manager controller pod not found in namespace $namespace"
+        fi
+    done
+    condition="oc -n ${namespace} get po --no-headers --ignore-not-found | egrep 'Running|Completed|Succeeded' | grep ^${name}"
+    wait_message="Waiting for pod ${name} in namespace ${namespace} to be running ..."
+    success_message="Pod ${name} in namespace ${namespace} is running."
+    error_message="Timeout after ${total_time_mins} minutes waiting for pod ${name} in namespace ${namespace} to be running."
+    wait_for_condition "${condition}" ${retries} ${sleep_time} "${wait_message}" "${success_message}" "${error_message}"
+    
+    #cainjector
+    name=$(${OC} get pod -n $namespace | (grep cert-manager-cainjector || echo "fail") | awk '{print $1}')
+    check_retries=10
+    while [[ $name == "fail" ]] && [ $check_retries -gt 0 ]; do
+        sleep 15
+        name=$(${OC} get pod -n $namespace | (grep cert-manager-cainjector || echo "fail") | awk '{print $1}')
+        debug1 "cert manager cainjector pod present: $name"
+        check_retries=$(( check_retries - 1 ))
+        if [ $check_retries -eq 0 ]; then
+            error "Cert manager cainjector pod not found in namespace $namespace"
+        fi
+    done
+    condition="oc -n ${namespace} get po --no-headers --ignore-not-found | egrep 'Running|Completed|Succeeded' | grep ^${name}"
+    wait_message="Waiting for pod ${name} in namespace ${namespace} to be running ..."
+    success_message="Pod ${name} in namespace ${namespace} is running."
+    error_message="Timeout after ${total_time_mins} minutes waiting for pod ${name} in namespace ${namespace} to be running."
+    wait_for_condition "${condition}" ${retries} ${sleep_time} "${wait_message}" "${success_message}" "${error_message}"
```

Comment:
I think if we already did cert-manager delegation at CP 2.0 level, It is possible that community or Red Hat is already installed in the cluster. As a result, we will not see any cert-manager operand pods in either `ibm-common-services` or control namespace.

IMHO I feel that cert-manager-operator pod check would be sufficient. But looking for other's opinion. FYI @bitscuit

Reply:
True, we cannot assume all the cert-manager pods will be there because of global cert-manager. We can probably add in more complex checks like checking for cert-manager pods cluster wide, but I think just checking for cert-manager-operator pod is good enough for the immediate issue

Reply:
We would only be waiting for cert manager if we remove the cert manager CR from the MASTER_NS. If cert manager is installed any other way, CERT_MANAGER_MIGRATED="false" and we would not call the wait_for_certmanager function

Reply:
As far as what we should check, from the slack thread, cpd is specifically looking for controller deployment to be ready so they may want more than just the operator pod ready. I think they want the script to fully wait for cert manager to deploy since they run this in sequence with other steps

Reply:
CertManager CR always exist in Bedrock, in both 2.0 and 3.0. And the cert-manager-operator pod also always exist. Whether the operand pods exist or not is dependent on whether the cluster has a global cert-manager or not.

So cannot always check for cert-manager operand pods. You can conditionally check them, but it would have to be generalized to checking cert-manager pods in all namespaces